sdk/services/kwentaToken.ts

```diff
@@ -0,0 +1,153 @@
+		if (!this.sdk.context.contracts.StakingRewards) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		const { hash } = await this.sdk.transactions.createContractTxn(
+			this.sdk.context.contracts.StakingRewards,
+			action,
+			[wei(amount).toBN()]
+		);
+
+		return hash;
+	}
+
+	public async approveLPToken() {
+		const StakingRewards = this.sdk.context.contracts.StakingRewards;
+		const KwentaArrakisVault = this.sdk.context.contracts.KwentaArrakisVault;
+
+		if (!StakingRewards || !KwentaArrakisVault) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		const { hash } = await this.sdk.transactions.createContractTxn(KwentaArrakisVault, 'approve', [
+			StakingRewards.address,
+			ethers.constants.MaxUint256,
+		]);
+
+		return hash;
+	}
+
+	public async getEarnDetails() {
+		const { StakingRewards, KwentaArrakisVault } = this.sdk.context.contracts;
+
+		if (!StakingRewards || !KwentaArrakisVault) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		const StakingRewardsContract = new EthCallContract(StakingRewards.address, StakingRewardsABI);
+		const KwentaArrakisVaultContract = new EthCallContract(
+			KwentaArrakisVault.address,
+			KwentaArrakisVaultABI
+		);
+
+		const [
+			balance,
+			earned,
+			periodFinish,
+			rewardRate,
+			totalSupply,
+			lpTokenBalance,
+			allowance,
+		]: ethers.BigNumber[] = await this.sdk.context.multicallProvider.all([
+			StakingRewardsContract.balanceOf(this.sdk.context.walletAddress),
+			StakingRewardsContract.earned(this.sdk.context.walletAddress),
+			StakingRewardsContract.periodFinish(),
+			StakingRewardsContract.rewardRate(),
+			StakingRewardsContract.totalSupply(),
+			KwentaArrakisVaultContract.balanceOf(this.sdk.context.walletAddress),
+			KwentaArrakisVaultContract.allowance(this.sdk.context.walletAddress, StakingRewards.address),
+		]);
+
+		return {
+			balance: wei(balance),
+			earned: wei(earned),
+			endDate: periodFinish.toNumber(),
+			rewardRate: wei(rewardRate),
+			totalSupply: wei(totalSupply),
+			lpTokenBalance: wei(lpTokenBalance),
+			allowance: wei(allowance),
+		};
+	}
+
+	public async claimRewards() {
+		const StakingRewards = this.sdk.context.contracts.StakingRewards;
+
+		if (!StakingRewards) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		const { hash } = await this.sdk.transactions.createContractTxn(StakingRewards, 'getReward', []);
+
+		return hash;
+	}
+
+	public async getStakingData() {
+		const {
+			RewardEscrow,
+			StakingRewards,
+			KwentaToken,
+			SupplySchedule,
+			vKwentaToken,
+			MultipleMerkleDistributor,
+			veKwentaToken,
+		} = this.sdk.context.contracts;
+
+		if (
+			!RewardEscrow ||
+			!StakingRewards ||
+			!KwentaToken ||
+			!SupplySchedule ||
+			!vKwentaToken ||
+			!MultipleMerkleDistributor ||
+			!veKwentaToken
+		) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		const RewardEscrowContract = new EthCallContract(RewardEscrow.address, []);
```

Comment:
Do you think it's worth also initialising EthCall contracts along with the typed contracts? Also did you get a chance to see if that fork was any good for combining them?

Reply:
I think it's probably worth a shot. Maybe we only do that for the contracts we use with multicall though?